exercises/04-exercise.py

```diff
@@ -0,0 +1,220 @@
+	for file_name in an_files_only:
+		an_df = pd.read_csv((directory_string + "/" + file_name) , delimiter=" ", names = ['time (s)', 'time (ns)', 'index', 'counts'], header = None)
+		an_df['volts'] = an_df['counts'].apply(adc2counts)
+		an_df['total time (s)'] = an_df['time (s)'] + 1e-9*an_df['time (ns)']
+		an_data[file_name] = an_df
+
+		temp_df.append(an_df)
+		
+	return an_data, temp_df
+
+def new_an_dfs(an_df):
+	'''
+	returns a new dataframe for an_files with only
+	two columns --> col0 = total time (s) col1 = counts
+	'''
+	new_an_df = []
+	for df in an_df:
+		new_df = df[['total time (s)' , 'counts']].copy()	
+		new_df['counts'] = new_df['counts'].apply(adc2counts)
+		new_an_df.append(new_df)
+	return new_an_df
+
+def en_files_to_dict(en_files_only):
+	'''
+	creates a dictionary in which key = filename and the value = dataframe
+	associated with that specific file
+	returns en_dictionary and dataframes(stored in list) for each file
+	'''
+	en_data = dict()
+	temp_df = []
+	for file_name in en_files_only:
+		en_df = pd.read_csv(directory_string + "/" + file_name, delimiter=" ", names = ['time (s)', 'time (ns)', 'encoder', 'index', 'di'], header = None)
+		en_df['total time (s)'] = en_df['time (s)'] + 1e-9 * en_df['time (ns)']
+		encoder_to_energy = en_df['encoder'].apply(enc2counts)
+		en_df['energy'] = encoder_to_energy
+		en_data[file_name] = en_df
+		temp_df.append(en_df)
+		
+	return en_data, temp_df
+
+def dev_names(dev_names_path):
+	'''
+	reads in a text file and finds the device name along 
+	with the energy 
+	returns a dictionary with >> key = device name , value = energy
+	'''
+	dev_names = dict()
+	with open(dev_names_path) as file:
+		for line in file:
+			if not line.startswith("#"):
+				temp_split = line.split(":",1)
+				if len(temp_split) > 1:
+					dev_names[temp_split[0].strip()] = temp_split[1].strip()
+	return dev_names
+
+def manipulate_en_df(en_df):
+	'''
+	creating a new dataframe with only two columns
+	col0 = total time (s)- col1 = energy
+	returns a list with the new dataframe from each en_file 
+	'''
+	en_df_new = []
+	for df in en_df:
+		new_en_df = df[['total time (s)' , 'energy']].copy()	
+		en_df_new.append(new_en_df)
+	return en_df_new
+	
+directory_string = "/home/jdiaz/projects/data-monitoring/data/iss_sample_data"
+dev_names_path = "/home/jdiaz/projects/data-monitoring/data/iss_sample_data/devnames.txt"
+
+'''
+going to store each file name in a list
+index 2 indicates that we only want filenames and no directory path or directory names
+'next' grabs the next value in iterator 
+'''
+files_only = next(os.walk(directory_string))[2] 
+
+txt_files_only = [file for file in files_only if file.endswith(".txt")]
+an_files_only = [file for file in files_only if file.startswith("an_")]	
+en_files_only = [file for file in files_only if file.startswith("en_")]
+
+#all files without .txt extension
+raw_files = [key for key in files_only if not key.endswith(".txt")]
+
+raw_files_read_in = raw_read_in(raw_files)
```

Comment:
why are you making it an array? It should not be needed

Reply:
im saving the data of each file in an array, so index 0 would contain the data for an_file[0] ... is that okay?